lib/longtail/stackoverflow/posts.pl

```diff
@@ -75,7 +75,8 @@
                 # 1 = question, 2 = answer, 3 might be external
                 my $link_type_id = $4;
                 #warn "$id -> $post_id -> $related_post_id -> $link_type_id\n";
-                push (@{$post_links{$post_id}} ,$related_post_id);
+                $post_links{$post_id}{$related_post_id} = 1;
+                $post_links{$related_post_id}{$post_id} = 1;
```

Comment:
Down below [L333](https://github.com/duckduckgo/zeroclickinfo-longtail/blob/a8599558cae06c79e906f6940b6425bd22b96fac/lib/longtail/stackoverflow/posts.pl#L333), you'll want to take the `keys` of the `$parent_id` now to create the array ref.